File: 1411_Number_of_Ways_to_Paint_N_x_3_Grid/solution.c
int numOfWays(int n) {
    long MOD = 1000000007;
    long same = 6, diff = 6;

    for (int i = 1; i < n; i++) {
        long same_2 = (same * 2 + diff * 2) % MOD;
        long diff_2 = (same * 2 + diff * 3) % MOD;
        same = same_2; diff = diff_2;
    }
    return (int)((same + diff) % MOD);
}

File: 1411_Number_of_Ways_to_Paint_N_x_3_Grid/solution.py
class Solution:
    def numOfWays(self, n: int) -> int:
        MOD = 10**9 + 7
        same = diff = 6
        
        for i in range(1, n):
            same, diff = (2 * same + 2 * diff) % MOD, (2 * same + 3 * diff) % MOD
            
        return (same + diff) % MOD
